03 - Clustering Misorientations.ipynb

```diff
@@ -165,120 +246,296 @@
+    "sigma13a = Rotation.from_neo_euler(AxAngle.from_axes_angles((1, 0, 0), np.radians(76.89)))\n",
+    "sigma13b = Rotation.from_neo_euler(AxAngle.from_axes_angles((2, 1, 0), np.radians(57.22)))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# From Laine et al\n",
+    "sigma7a = Rotation.from_neo_euler(AxAngle.from_axes_angles((1, 0, 0), np.radians(64.40)))\n",
+    "sigma11a = Rotation.from_neo_euler(AxAngle.from_axes_angles((1, 0, 0), np.radians(34.96)))\n",
+    "sigma11b = Rotation.from_neo_euler(AxAngle.from_axes_angles((1, 1, 0), np.radians(85.03)))\n",
+    "sigma13a = Rotation.from_neo_euler(AxAngle.from_axes_angles((1, 0, 0), np.radians(76.89)))\n",
+    "sigma13b = Rotation.from_neo_euler(AxAngle.from_axes_angles((1, 1, 0), np.radians(57.22)))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# From Bonnet et al\n",
+    "sigma7a = Rotation.from_neo_euler(AxAngle.from_axes_angles((2, 1, 0), np.radians(64.62)))\n",
+    "sigma11a = Rotation.from_neo_euler(AxAngle.from_axes_angles((2, 1, 0), np.radians(35.10)))\n",
+    "sigma11b = Rotation.from_neo_euler(AxAngle.from_axes_angles((1, 0, 0), np.radians(84.78)))\n",
+    "sigma13a = Rotation.from_neo_euler(AxAngle.from_axes_angles((2, 1, 0), np.radians(76.66)))\n",
+    "sigma13b = Rotation.from_neo_euler(AxAngle.from_axes_angles((1, 0, 0), np.radians(57.42)))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "twin_theory = Orientation.stack([sigma7a, sigma11a, sigma11b, sigma13a, sigma13b]).flatten()"
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "<h2><center> Plotting our results </h2></center>"
+    "Calculate difference, defined as minimum rotation angle, between measured and theoretical values"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "difference = Misorientation((~cluster_means).outer(twin_theory)).set_symmetry(D6, D6).angle.data"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 25,
+   "execution_count": null,
    "metadata": {},
-   "outputs": [
-    {
-     "data": {
-      "text/plain": [
-       "<matplotlib.legend.Legend at 0x7fc63d024588>"
-      ]
-     },
-     "execution_count": 25,
-     "metadata": {},
-     "output_type": "execute_result"
-    }
-   ],
+   "outputs": [],
+   "source": [
+    "np.degrees(difference)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mismisori = (~twin_theory).outer(cluster_means)\n",
+    "mismisori_equiv = D6.outer(~twin_theory).outer(D6).outer(D6).outer(cluster_means).outer(D6)\n",
```

Comment:
I'm pretty sure this is the correct idea.

Reply:
Yeah, but this is the one that still seems to have an issue with order...